### PR TITLE
Add blog table of contents and anchor navigation

### PR DESCRIPTION
**Pull request by colebow**
I'm gonna make y'all test this yourself. There's probably room to make this a little fancier/spiffier, but this at least gets everything added to the site and functioning.

Personally, I'm actually not sure I love the table of contents - for a lot of blog posts, it's giving us virtually no value (see: all the recaps where we're using two headers). But the anchors are nice.

### CODE REVIEW

**Comment by martint** [comment]
It looks very broken on mobile, btw

**Comment by martint** [comment]
7DA777F1-8B3E-45E6-A02F-51F6F6BECAF1

**Comment by mosabua** [comment]
I still have to look at it but for mobile we should just not show that div that contains the sidenav.

**Comment by colebow** [comment]
Learned what bootstrap is, should now hide the table of contents for screens <970px wide.

**Comment by bitsondatadev** [comment]
https://user-images.githubusercontent.com/8547669/217982978-6bd0cba5-9b3d-4162-986c-b87601cf6c0c.MOV


This looks pretty good IMO!

**Comment by colebow** [comment]
> Upload.from.GitHub.for.iOS.MOV 
> This looks pretty good IMO!

One issue I notice here is that a long table of contents on a short screen (e.g. a phone rotated sideways...) means you can't see the bottom of the table of contents without getting to the bottom of the blog. I might try to look into fixing that with a different sticky property, but IMO that's not a requirement, as nearly all blogs don't have TOCs that long, and I imagine the number of people reading the Trino blog sideways on their phone is near zero.

**Comment by colebow** [comment]
Nevermind, figured out how to make it work. I enjoy CSS properties.

File: _layouts/blog.html [hunk 1]
```diff
@@ -36,7 +36,25 @@ <h1><a href="/blog/">Trino blog</a></h1>
...
+  {% else %}
+    <div class="row">
+      <div class="col-md-8 col-sm-12">
+        <div class="mx-2">
+          {% include anchor_headings.html html=content anchorBody="#" %}
+        </div>
+      </div>
+      <div class="col-md-4 d-md-block d-none">
+        <div style="position: sticky; position:-webkit-sticky; top:10px; max-height:100vh; overflow-y:auto;" class="outline-card">
+          <u>Table of Contents</u>
```
Comment (mosabua) [inline _layouts/blog.html:51]:
```suggestion
          <u>Table of contents</u>
```

File: _layouts/blog.html [hunk 2]
```diff
@@ -36,7 +36,25 @@ <h1><a href="/blog/">Trino blog</a></h1>
     </div>
   </div>
 
-  {{ content }}
+  <!-- Avoids showing table of contents on blog index page -->
+  {% if content contains '<!-- Blog Index -->' %}
```
Comment (mosabua) [inline _layouts/blog.html:40]: Lets make this independent of the page we use it on by using a value like "no-right-nav"  or so

**Review by mosabua — CHANGES_REQUESTED** [review]
Two minor nits .. but then really nice and good to go. We might want to go back to a bunch of older blogs and check headers and add some or fix level of them. And adding it to the TCB pages would also be cool.

File: _layouts/blog.html (hunk 2 above)
Comment (colebow) [reply _layouts/blog.html:40]: IMO that's an easy change to make if/when we decide we want to remove the nav from a post at some point. But if it's only one page for now, we can call that page out specifically for better clarity.

Comment (mosabua) [reply _layouts/blog.html:40]: ok

**Review by mosabua — APPROVED** [review]
Love it.. lets get this live!

**Pull request merged**